test/index.test.ts

```diff
@@ -146,46 +144,95 @@ describe("Testing Toucan-SDK contract interactions", function () {
         wethBalance: await weth.balanceOf(addr1.address),
       });
 
-      const cost = await toucan.calculateNeededTokenAmount(
+      const cost = await toucan.calculateExpectedPoolTokenForToken(
+        weth.address,
         "NCT",
-        ONE_ETHER,
-        weth
+        ONE_ETHER
       );
-      await toucan.autoOffsetUsingSwapToken("NCT", ONE_ETHER, weth);
+
+      await toucan.autoOffsetExactInToken(network.weth, "NCT", ONE_ETHER);
 
       state.push({
         nctSupply: await pool.totalSupply(),
         wethBalance: await weth.balanceOf(addr1.address),
       });
 
       expect(
-        formatEther(state[0].nctSupply.sub(ONE_ETHER)),
-        `Expect NCT supply to be less by 1.0`
+        formatEther(state[0].nctSupply.sub(cost)),
+        `Expect NCT supply to be less by ${formatEther(cost)}`
       ).to.be.equal(formatEther(state[1].nctSupply));
 
       expect(
-        formatEther(state[0].wethBalance.sub(cost)),
-        `Expect addr1's WETH balance to be less by ${formatEther(cost)}`
+        formatEther(state[0].wethBalance.sub(ONE_ETHER)),
+        `Expect addr1's WETH balance to be less by 1.0`
       ).to.be.equal(formatEther(state[1].wethBalance));
     });
 
-    it("Should retire 1 TCO2 using ETH deposit", async function () {
+    it("Should retire 1 TCO2 using ETH (native token) #autoOffsetExactInETH)", async function () {
+      const pool = await toucan.getPoolContract("NCT");
+
+      const state: any[] = [];
```

Comment:
I've added some typing but had to keep an any otherwise throws some errors and I haven't been able to figure out how to solve this. pls check. maybe you have an idea. otherwise, can we please get this closed?